components/Schedule.js

```diff
@@ -0,0 +1,346 @@
+        right: 0;
+        background: linear-gradient(153.43deg, rgba(0, 0, 0, 0) 0%, #000000 100%);
+    }
+`;
+
+const SpeakerInfo2 = styled.div`
+    height: 140px;
+    width: 320px;
+    position: absolute;
+    bottom: 0;
+    right: 0px;
+    text-align: right;
+    color: #dfdfdf;
+    span {
+        display: block;
+        position: absolute;
+        right: 120px;
+        z-index: 50;
+        top: 85px;
+        font-size: 28px;
+        line-height: 40px;
+        font-family: Teko;
+        text-transform: uppercase;
+    }
+    @media (min-width: 320px) and (max-width: 520px) {
+        width: 100%;
+        padding-top: 20px;
+        position: relative;
+        text-align: right;
+
+        span {
+            top: 100px;
+            right: 60px;
+        }
+    }
+`;
+
+const Row = styled.div`
+    display: flex;
+    flex-direction: column;
+    max-width: 540px;
+    margin: 100px auto;
+    @media (min-width: 1000px) {
+        max-width: 980px;
+        flex-direction: row;
+    }
+    @media (min-width: 1200px) {
+        max-width: 1080px;
+        flex-direction: row;
+    }
+`;
+
+const Col = styled.div`
+    flex 1;
+        
+`;
+
+const Slot = ({title, image, speaker, startTime, endTime, dark}) => {
+    let content;
+    let timeString = '';
+    if (startTime) {
+        timeString += startTime;
+    }
+    if (endTime) {
+        timeString += ` - ${endTime}`;
+    }
+    if (title) {
+        content = <Title image={image}>{title}</Title>;
+    } else if (speaker) {
+        let speaker2;
+        const {coSpeaker} = speaker;
+        if (coSpeaker) {
+            speaker2 = (
+                <SpeakerInfo2>
+                    <img src={coSpeaker.imageUrls.small} alt={coSpeaker.name} />
+                    <span>{coSpeaker.name}</span>
+                    <ImageOverlay2 />
+                </SpeakerInfo2>
+            );
+        }
+        content = (
+            <Speaker>
+                <img src={speaker.imageUrls.small} alt={speaker.name} />
+                <ImageOverlay />
+                <SpeakerInfo>
+                    {speaker.name}
+                    <div dangerouslySetInnerHTML={{__html: speaker.teaser}} />
+                </SpeakerInfo>
+                {speaker2}
+            </Speaker>
+        );
+    }
+
+    return (
+        <Block dark={dark}>
+            <TimeInfo>{timeString}</TimeInfo>
+            {content}
+        </Block>
+    );
+};
+
+Slot.propTypes = {
+    title: PropTypes.string,
+    image: PropTypes.string,
+    speaker: PropTypes.object,
+    startTime: PropTypes.string,
+    endTime: PropTypes.string,
+    dark: PropTypes.bool,
+};
+
+export default class Schedule extends React.PureComponent {
+    constructor(props) {
+        super(props);
+
+        this.state = {slots: this.parseSlots()};
+    }
+    static propTypes = {
+        speakers: PropTypes.array.isRequired,
+    };
+
+    parseSlots = () => {
+        const slots = new Array(15);
```

Comment:
Do you feel that it is necessary to create the number of speakers here already? Why not just `slots = []`?

Reply:
just to make shure i can access all the slots[number]